### PR TITLE
Fix `IsNullOrEmptyStateTrigger` for Uno to consider FEs as non-empty

### PR DESCRIPTION
**Pull request by Youssef1313**
## Fixes

Fixes https://github.com/unoplatform/uno/issues/17311

## PR Type

What kind of change does this PR introduce?

## What is the current behavior?

`FrameworkElement`s in Uno are incorrectly `IEnumerable`s, so they could be considered as empty incorrectly.

## PR Checklist

Please check if your PR fulfills the following requirements: 

- [ ] Created a feature/dev branch in your fork (vs. submitting directly from a commit on main)
- [ ] Based off latest main branch of toolkit
- [ ] Tested code with current supported SDKs
- [ ] New component
  - [ ] Documentation has been added
  - [ ] Sample in sample app has been added
  - [ ] Analyzers are passing for documentation and samples
  - [ ] Icon has been created (if new sample) following the [Thumbnail Style Guide and templates](https://github.com/CommunityToolkit/WindowsCommunityToolkit-design-assets)
- [ ] Tests for the changes have been added (if applicable)
- [ ] Header has been added to all new source files
- [ ] Contains **NO** breaking changes

### CODE REVIEW

File: components/Triggers/src/IsNullOrEmptyStateTrigger.cs [hunk 1]
```diff
@@ -118,7 +118,11 @@ private static bool IsNullOrEmpty(object val)
 
         // Object is not an ICollection, check for an empty IEnumerable
         var valEnumerable = val as IEnumerable;
-        if (valEnumerable != null)
+        if (val is IEnumerable valEnumerable
+#if HAS_UNO
```
Comment (Arlodotexe) [inline components/Triggers/src/IsNullOrEmptyStateTrigger.cs:122]:
```suggestion
// Workaround to regression introduced in https://github.com/unoplatform/uno/pull/16834
// Track https://github.com/unoplatform/uno/issues/17311 for cleanup
#if HAS_UNO
```

**Review by Arlodotexe — APPROVED** [review]

**Comment by Arlodotexe** [comment]
Thanks for the contribution @Youssef1313, I've added tracking details in a code comment so we can cleanup the workaround once the regression is fixed.

**Comment by Arlodotexe** [comment]
Seems these changes don't build as expected:
```
         /home/runner/work/Windows/Windows/components/Triggers/src/IsNullOrEmptyStateTrigger.cs(121,32): error CS0128: A local variable or function named 'valEnumerable' is already defined in this scope [/home/runner/work/Windows/Windows/components/Triggers/src/CommunityToolkit.WinUI.Triggers.csproj::TargetFramework=net8.0]
         /home/runner/work/Windows/Windows/components/Triggers/src/IsNullOrEmptyStateTrigger.cs(127,34): error CS8602: Dereference of a possibly null reference. [/home/runner/work/Windows/Windows/components/Triggers/src/CommunityToolkit.WinUI.Triggers.csproj::TargetFramework=net8.0]
```

**Comment by agneszitte** [comment]
Thank you for the review @Arlodotexe!

**Comment by agneszitte** [comment]
@Arlodotexe, @Youssef1313 seems that some tests are failing

**Comment by Youssef1313** [comment]
The failure looks unrelated to the change

**Pull request merged**